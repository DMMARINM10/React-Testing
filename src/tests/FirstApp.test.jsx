import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Tests on FirstApp component', () => {
    test('FirstApp component should be equal to snapshot', () => { 
        //render component in memory
        const title = 'Hola Mundo'
        const { container } = render(<FirstApp title={title}/>)
        expect(container).toMatchSnapshot()
    })

    test('title prop should be printed in Component', () => {
        const title = 'Hola Mundo'
        //get node by text
        //getBy just bring one element and throw error if there are multiple, in that case gettAllBy (array)
        const { container, getByText, getByTestId } = render(<FirstApp title={title}/>)
        expect(getByText(title)).toBeTruthy() //it is found, it exists

        //not used as much
        const h1 = container.querySelector('h1')
        expect(h1.innerHTML).toContain(title)

        //screen es el objecto que recibo del render dentro del body
        //renderizado del comp actualizado luego de cambios en el DOM
        expect(screen.getByRole('heading', { level: 1 } ).innerHTML).toContain(title)

        expect(getByTestId('title').innerHTML).toBe(title)
    })
})