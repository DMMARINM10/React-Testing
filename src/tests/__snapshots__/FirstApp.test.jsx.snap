// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on FirstApp component FirstApp component should be equal to snapshot 1`] = `
<div>
  <h1
    data-testid="title"
  >
    Hola Mundo
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
